### PR TITLE
Added support for z-locked cylindrical billboarding

### PR DESCRIPTION
**Pull request by fallenoak**
Not all billboarding is spherical. This adds support for z-axis cylindrical billboarding.

Note that this requires https://github.com/wowserhq/blizzardry/pull/92 to be merged.

### CODE REVIEW

File: src/lib/pipeline/m2/index.js [hunk 1]
```diff
@@ -377,11 +379,21 @@ class M2 extends THREE.Group {
...
+      switch (bone.userData.billboardType) {
+        case 0:
+          this.applySphericalBillboard(camera, bone);
+          break;
+        case 3:
+          this.applyCylindricalZBillboard(camera, bone);
+          break;
+        default:
+          break;
+      }
```
Comment (timkurvers) [inline src/lib/pipeline/m2/index.js]: Assuming we'll be adding support for X and Y once we find some models that actually have it? :smile:

Comment (fallenoak) [inline src/lib/pipeline/m2/index.js]: Bingo!

At some point, it'd be sweet if we could define some kind of search API for the pipeline to let us find things like that. I'm sure it'd be painfully slow, but it'd be better than having to manually hunt for things.

Comment (timkurvers) [inline src/lib/pipeline/m2/index.js]: Yeah, that'd be super cool, does sound slow :smile:

**Comment by timkurvers** [comment]
:tada:

**Pull request merged**